### DISCUSSION TITLE
DataBinding에서 BindingAdapter 사용

### DISCUSSION THREAD

**Discussion by choius323** [boostcampwm-2022/android08-KiwiTalk#25] [Show and tell]
비슷한 기능을 하는데 어떤 게 더 나을까요?

```kotlin
android:text="@{chatList.extraData.getOrDefault(Const.MAP_KEY_COUNTRY, `지역`).toString()}"
```
이렇게 뷰마다 각각 데이터를 불러와서 직접 넣어주는 방식이 있습니다.
BindingAdapter가 따로 필요 없지만, xml에서 코드가 길어지고 보기 어렵습니다.

```kotlin
@BindingAdapter(value = ["chatList", "map_key"])
fun setTextWithList(view: TextView, chatList: Channel, key: String) {
    val default: Any = when (key) {
        Const.MAP_KEY_KEYWORD -> "키워드"
        Const.MAP_KEY_COUNTRY -> "지역"
        else -> "."
    }
    val value = chatList.getExtraValue(key, default)

    view.text = value.toString()
}
```
```xml
app:map_key="@{Const.MAP_KEY_KEYWORD}"
app:chatList="@{chatList}"
```
뷰에서 데이터를 넣어주면 함수에서 처리해주는 방식이 있습니다.
xml이 깔끔하지만, 파라미터 2개가 필요하고 함수가 길어지는 단점이 있습니다.(enum class 사용하면 조금 줄어들 수 있습니다.)

</br>
</br>

지금은 키워드, 지역만 있고 더 추가될 내용이 있을지 떠오르는 건 없네요.

**Comment by KimGyeongyeon** [comment]
바인딩 어댑터 사용시 xml에서 chat list와 map key의 관계가 잘 보이지 않는 것 같습니다. 
List -> String 변환이 크게 확장될 일이 없을 것 같아서 저라면 xml상에서 해결항 것 같아요
